// @stacks/blockchain-api-client/src/generated/models/AverageBlockTimesResponse.ts
/* tslint:disable */
/* eslint-disable */
/**
 * Stacks Blockchain API
 * Welcome to the API reference overview for the [Stacks Blockchain API](https://docs.hiro.so/stacks-blockchain-api).  [Download Postman collection](https://hirosystems.github.io/stacks-blockchain-api/collection.json) 
 *
 * The version of the OpenAPI document: STACKS_API_VERSION
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Request to fetch average block times (in seconds)
 * @export
 * @interface AverageBlockTimesResponse
 */
export interface AverageBlockTimesResponse {
    /**
     * Average block times over the last hour (in seconds)
     * @type {number}
     * @memberof AverageBlockTimesResponse
     */
    last_1h: number;
    /**
     * Average block times over the last 24 hours (in seconds)
     * @type {number}
     * @memberof AverageBlockTimesResponse
     */
    last_24h: number;
    /**
     * Average block times over the last 7 days (in seconds)
     * @type {number}
     * @memberof AverageBlockTimesResponse
     */
    last_7d: number;
    /**
     * Average block times over the last 30 days (in seconds)
     * @type {number}
     * @memberof AverageBlockTimesResponse
     */
    last_30d: number;
}

export function AverageBlockTimesResponseFromJSON(json: any): AverageBlockTimesResponse {
    return AverageBlockTimesResponseFromJSONTyped(json, false);
}

export function AverageBlockTimesResponseFromJSONTyped(json: any, ignoreDiscriminator: boolean): AverageBlockTimesResponse {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'last_1h': json['last_1h'],
        'last_24h': json['last_24h'],
        'last_7d': json['last_7d'],
        'last_30d': json['last_30d'],
    };
}

export function AverageBlockTimesResponseToJSON(value?: AverageBlockTimesResponse | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'last_1h': value.last_1h,
        'last_24h': value.last_24h,
        'last_7d': value.last_7d,
        'last_30d': value.last_30d,
    };
}
