// @stacks/blockchain-api-client/src/generated/models/FtHolderEntry.ts
/* tslint:disable */
/* eslint-disable */
/**
 * Stacks Blockchain API
 * Welcome to the API reference overview for the [Stacks Blockchain API](https://docs.hiro.so/stacks-blockchain-api).  [Download Postman collection](https://hirosystems.github.io/stacks-blockchain-api/collection.json) 
 *
 * The version of the OpenAPI document: STACKS_API_VERSION
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * 
 * @export
 * @interface FtHolderEntry
 */
export interface FtHolderEntry {
    /**
     * 
     * @type {string}
     * @memberof FtHolderEntry
     */
    address: string;
    /**
     * 
     * @type {string}
     * @memberof FtHolderEntry
     */
    balance: string;
}

export function FtHolderEntryFromJSON(json: any): FtHolderEntry {
    return FtHolderEntryFromJSONTyped(json, false);
}

export function FtHolderEntryFromJSONTyped(json: any, ignoreDiscriminator: boolean): FtHolderEntry {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'address': json['address'],
        'balance': json['balance'],
    };
}

export function FtHolderEntryToJSON(value?: FtHolderEntry | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'address': value.address,
        'balance': value.balance,
    };
}
