// @stacks/blockchain-api-client/src/generated/models/FungibleTokenHolderList.ts
/* tslint:disable */
/* eslint-disable */
/**
 * Stacks Blockchain API
 * Welcome to the API reference overview for the [Stacks Blockchain API](https://docs.hiro.so/stacks-blockchain-api).  [Download Postman collection](https://hirosystems.github.io/stacks-blockchain-api/collection.json) 
 *
 * The version of the OpenAPI document: STACKS_API_VERSION
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    FtHolderEntry,
    FtHolderEntryFromJSON,
    FtHolderEntryFromJSONTyped,
    FtHolderEntryToJSON,
} from './';

/**
 * List of Fungible Token holders
 * @export
 * @interface FungibleTokenHolderList
 */
export interface FungibleTokenHolderList {
    /**
     * The number of holders to return
     * @type {number}
     * @memberof FungibleTokenHolderList
     */
    limit: number;
    /**
     * The number to holders to skip (starting at `0`)
     * @type {number}
     * @memberof FungibleTokenHolderList
     */
    offset: number;
    /**
     * The number of holders available
     * @type {number}
     * @memberof FungibleTokenHolderList
     */
    total: number;
    /**
     * The total supply of the token (the sum of all balances)
     * @type {string}
     * @memberof FungibleTokenHolderList
     */
    total_supply: string;
    /**
     * 
     * @type {Array<FtHolderEntry>}
     * @memberof FungibleTokenHolderList
     */
    results: Array<FtHolderEntry>;
}

export function FungibleTokenHolderListFromJSON(json: any): FungibleTokenHolderList {
    return FungibleTokenHolderListFromJSONTyped(json, false);
}

export function FungibleTokenHolderListFromJSONTyped(json: any, ignoreDiscriminator: boolean): FungibleTokenHolderList {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'limit': json['limit'],
        'offset': json['offset'],
        'total': json['total'],
        'total_supply': json['total_supply'],
        'results': ((json['results'] as Array<any>).map(FtHolderEntryFromJSON)),
    };
}

export function FungibleTokenHolderListToJSON(value?: FungibleTokenHolderList | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'limit': value.limit,
        'offset': value.offset,
        'total': value.total,
        'total_supply': value.total_supply,
        'results': ((value.results as Array<any>).map(FtHolderEntryToJSON)),
    };
}
