// @stacks/blockchain-api-client/src/generated/models/PoxCycleSignerStackersListResponse.ts
/* tslint:disable */
/* eslint-disable */
/**
 * Stacks Blockchain API
 * Welcome to the API reference overview for the [Stacks Blockchain API](https://docs.hiro.so/stacks-blockchain-api).  [Download Postman collection](https://hirosystems.github.io/stacks-blockchain-api/collection.json) 
 *
 * The version of the OpenAPI document: STACKS_API_VERSION
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    PoxStacker,
    PoxStackerFromJSON,
    PoxStackerFromJSONTyped,
    PoxStackerToJSON,
} from './';

/**
 * GET request that returns stackers for a signer in a PoX cycle
 * @export
 * @interface PoxCycleSignerStackersListResponse
 */
export interface PoxCycleSignerStackersListResponse {
    /**
     * The number of stackers to return
     * @type {number}
     * @memberof PoxCycleSignerStackersListResponse
     */
    limit: number;
    /**
     * The number to stackers to skip (starting at `0`)
     * @type {number}
     * @memberof PoxCycleSignerStackersListResponse
     */
    offset: number;
    /**
     * The total number of stackers
     * @type {number}
     * @memberof PoxCycleSignerStackersListResponse
     */
    total: number;
    /**
     * 
     * @type {Array<PoxStacker>}
     * @memberof PoxCycleSignerStackersListResponse
     */
    results: Array<PoxStacker>;
}

export function PoxCycleSignerStackersListResponseFromJSON(json: any): PoxCycleSignerStackersListResponse {
    return PoxCycleSignerStackersListResponseFromJSONTyped(json, false);
}

export function PoxCycleSignerStackersListResponseFromJSONTyped(json: any, ignoreDiscriminator: boolean): PoxCycleSignerStackersListResponse {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'limit': json['limit'],
        'offset': json['offset'],
        'total': json['total'],
        'results': ((json['results'] as Array<any>).map(PoxStackerFromJSON)),
    };
}

export function PoxCycleSignerStackersListResponseToJSON(value?: PoxCycleSignerStackersListResponse | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'limit': value.limit,
        'offset': value.offset,
        'total': value.total,
        'results': ((value.results as Array<any>).map(PoxStackerToJSON)),
    };
}
