// @stacks/blockchain-api-client/src/generated/models/PoxCycleListResponse.ts
/* tslint:disable */
/* eslint-disable */
/**
 * Stacks Blockchain API
 * Welcome to the API reference overview for the [Stacks Blockchain API](https://docs.hiro.so/stacks-blockchain-api).  [Download Postman collection](https://hirosystems.github.io/stacks-blockchain-api/collection.json) 
 *
 * The version of the OpenAPI document: STACKS_API_VERSION
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    PoxCycle,
    PoxCycleFromJSON,
    PoxCycleFromJSONTyped,
    PoxCycleToJSON,
} from './';

/**
 * GET request that returns PoX cycles
 * @export
 * @interface PoxCycleListResponse
 */
export interface PoxCycleListResponse {
    /**
     * The number of cycles to return
     * @type {number}
     * @memberof PoxCycleListResponse
     */
    limit: number;
    /**
     * The number to cycles to skip (starting at `0`)
     * @type {number}
     * @memberof PoxCycleListResponse
     */
    offset: number;
    /**
     * The total number of cycles
     * @type {number}
     * @memberof PoxCycleListResponse
     */
    total: number;
    /**
     * 
     * @type {Array<PoxCycle>}
     * @memberof PoxCycleListResponse
     */
    results: Array<PoxCycle>;
}

export function PoxCycleListResponseFromJSON(json: any): PoxCycleListResponse {
    return PoxCycleListResponseFromJSONTyped(json, false);
}

export function PoxCycleListResponseFromJSONTyped(json: any, ignoreDiscriminator: boolean): PoxCycleListResponse {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'limit': json['limit'],
        'offset': json['offset'],
        'total': json['total'],
        'results': ((json['results'] as Array<any>).map(PoxCycleFromJSON)),
    };
}

export function PoxCycleListResponseToJSON(value?: PoxCycleListResponse | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'limit': value.limit,
        'offset': value.offset,
        'total': value.total,
        'results': ((value.results as Array<any>).map(PoxCycleToJSON)),
    };
}
