// @stacks/blockchain-api-client/src/generated/models/PoxCycle.ts
/* tslint:disable */
/* eslint-disable */
/**
 * Stacks Blockchain API
 * Welcome to the API reference overview for the [Stacks Blockchain API](https://docs.hiro.so/stacks-blockchain-api).  [Download Postman collection](https://hirosystems.github.io/stacks-blockchain-api/collection.json) 
 *
 * The version of the OpenAPI document: STACKS_API_VERSION
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * 
 * @export
 * @interface PoxCycle
 */
export interface PoxCycle {
    /**
     * 
     * @type {number}
     * @memberof PoxCycle
     */
    block_height: number;
    /**
     * 
     * @type {string}
     * @memberof PoxCycle
     */
    index_block_hash: string;
    /**
     * 
     * @type {number}
     * @memberof PoxCycle
     */
    cycle_number: number;
    /**
     * 
     * @type {number}
     * @memberof PoxCycle
     */
    total_weight: number;
    /**
     * 
     * @type {string}
     * @memberof PoxCycle
     */
    total_stacked_amount: string;
    /**
     * 
     * @type {number}
     * @memberof PoxCycle
     */
    total_signers: number;
}

export function PoxCycleFromJSON(json: any): PoxCycle {
    return PoxCycleFromJSONTyped(json, false);
}

export function PoxCycleFromJSONTyped(json: any, ignoreDiscriminator: boolean): PoxCycle {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'block_height': json['block_height'],
        'index_block_hash': json['index_block_hash'],
        'cycle_number': json['cycle_number'],
        'total_weight': json['total_weight'],
        'total_stacked_amount': json['total_stacked_amount'],
        'total_signers': json['total_signers'],
    };
}

export function PoxCycleToJSON(value?: PoxCycle | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'block_height': value.block_height,
        'index_block_hash': value.index_block_hash,
        'cycle_number': value.cycle_number,
        'total_weight': value.total_weight,
        'total_stacked_amount': value.total_stacked_amount,
        'total_signers': value.total_signers,
    };
}
