// @stacks/blockchain-api-client/src/generated/models/PoxCycleSignersListResponse.ts
/* tslint:disable */
/* eslint-disable */
/**
 * Stacks Blockchain API
 * Welcome to the API reference overview for the [Stacks Blockchain API](https://docs.hiro.so/stacks-blockchain-api).  [Download Postman collection](https://hirosystems.github.io/stacks-blockchain-api/collection.json) 
 *
 * The version of the OpenAPI document: STACKS_API_VERSION
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    PoxSigner,
    PoxSignerFromJSON,
    PoxSignerFromJSONTyped,
    PoxSignerToJSON,
} from './';

/**
 * GET request that returns signers for a PoX cycle
 * @export
 * @interface PoxCycleSignersListResponse
 */
export interface PoxCycleSignersListResponse {
    /**
     * The number of signers to return
     * @type {number}
     * @memberof PoxCycleSignersListResponse
     */
    limit: number;
    /**
     * The number to signers to skip (starting at `0`)
     * @type {number}
     * @memberof PoxCycleSignersListResponse
     */
    offset: number;
    /**
     * The total number of signers
     * @type {number}
     * @memberof PoxCycleSignersListResponse
     */
    total: number;
    /**
     * 
     * @type {Array<PoxSigner>}
     * @memberof PoxCycleSignersListResponse
     */
    results: Array<PoxSigner>;
}

export function PoxCycleSignersListResponseFromJSON(json: any): PoxCycleSignersListResponse {
    return PoxCycleSignersListResponseFromJSONTyped(json, false);
}

export function PoxCycleSignersListResponseFromJSONTyped(json: any, ignoreDiscriminator: boolean): PoxCycleSignersListResponse {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'limit': json['limit'],
        'offset': json['offset'],
        'total': json['total'],
        'results': ((json['results'] as Array<any>).map(PoxSignerFromJSON)),
    };
}

export function PoxCycleSignersListResponseToJSON(value?: PoxCycleSignersListResponse | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'limit': value.limit,
        'offset': value.offset,
        'total': value.total,
        'results': ((value.results as Array<any>).map(PoxSignerToJSON)),
    };
}
