// @stacks/blockchain-api-client/src/generated/models/PoxStacker.ts
/* tslint:disable */
/* eslint-disable */
/**
 * Stacks Blockchain API
 * Welcome to the API reference overview for the [Stacks Blockchain API](https://docs.hiro.so/stacks-blockchain-api).  [Download Postman collection](https://hirosystems.github.io/stacks-blockchain-api/collection.json) 
 *
 * The version of the OpenAPI document: STACKS_API_VERSION
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * 
 * @export
 * @interface PoxStacker
 */
export interface PoxStacker {
    /**
     * 
     * @type {string}
     * @memberof PoxStacker
     */
    stacker_address: string;
    /**
     * 
     * @type {string}
     * @memberof PoxStacker
     */
    stacked_amount: string;
    /**
     * 
     * @type {string}
     * @memberof PoxStacker
     */
    pox_address: string;
    /**
     * 
     * @type {string}
     * @memberof PoxStacker
     */
    stacker_type: PoxStackerStackerTypeEnum;
}

/**
* @export
* @enum {string}
*/
export enum PoxStackerStackerTypeEnum {
    solo = 'solo',
    pooled = 'pooled'
}

export function PoxStackerFromJSON(json: any): PoxStacker {
    return PoxStackerFromJSONTyped(json, false);
}

export function PoxStackerFromJSONTyped(json: any, ignoreDiscriminator: boolean): PoxStacker {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'stacker_address': json['stacker_address'],
        'stacked_amount': json['stacked_amount'],
        'pox_address': json['pox_address'],
        'stacker_type': json['stacker_type'],
    };
}

export function PoxStackerToJSON(value?: PoxStacker | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'stacker_address': value.stacker_address,
        'stacked_amount': value.stacked_amount,
        'pox_address': value.pox_address,
        'stacker_type': value.stacker_type,
    };
}
