// @stacks/blockchain-api-client/src/generated/models/PoxSigner.ts
/* tslint:disable */
/* eslint-disable */
/**
 * Stacks Blockchain API
 * Welcome to the API reference overview for the [Stacks Blockchain API](https://docs.hiro.so/stacks-blockchain-api).  [Download Postman collection](https://hirosystems.github.io/stacks-blockchain-api/collection.json) 
 *
 * The version of the OpenAPI document: STACKS_API_VERSION
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * 
 * @export
 * @interface PoxSigner
 */
export interface PoxSigner {
    /**
     * 
     * @type {string}
     * @memberof PoxSigner
     */
    signing_key: string;
    /**
     * The Stacks address derived from the signing_key.
     * @type {string}
     * @memberof PoxSigner
     */
    signer_address: string;
    /**
     * 
     * @type {number}
     * @memberof PoxSigner
     */
    weight: number;
    /**
     * 
     * @type {string}
     * @memberof PoxSigner
     */
    stacked_amount: string;
    /**
     * 
     * @type {number}
     * @memberof PoxSigner
     */
    weight_percent: number;
    /**
     * 
     * @type {number}
     * @memberof PoxSigner
     */
    stacked_amount_percent: number;
    /**
     * The number of solo stackers associated with this signer.
     * @type {number}
     * @memberof PoxSigner
     */
    solo_stacker_count: number;
    /**
     * The number of pooled stackers associated with this signer.
     * @type {number}
     * @memberof PoxSigner
     */
    pooled_stacker_count: number;
}

export function PoxSignerFromJSON(json: any): PoxSigner {
    return PoxSignerFromJSONTyped(json, false);
}

export function PoxSignerFromJSONTyped(json: any, ignoreDiscriminator: boolean): PoxSigner {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'signing_key': json['signing_key'],
        'signer_address': json['signer_address'],
        'weight': json['weight'],
        'stacked_amount': json['stacked_amount'],
        'weight_percent': json['weight_percent'],
        'stacked_amount_percent': json['stacked_amount_percent'],
        'solo_stacker_count': json['solo_stacker_count'],
        'pooled_stacker_count': json['pooled_stacker_count'],
    };
}

export function PoxSignerToJSON(value?: PoxSigner | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'signing_key': value.signing_key,
        'signer_address': value.signer_address,
        'weight': value.weight,
        'stacked_amount': value.stacked_amount,
        'weight_percent': value.weight_percent,
        'stacked_amount_percent': value.stacked_amount_percent,
        'solo_stacker_count': value.solo_stacker_count,
        'pooled_stacker_count': value.pooled_stacker_count,
    };
}
