// @stacks/blockchain-api-client/src/generated/models/RosettaBlockMetadata1.ts
/* tslint:disable */
/* eslint-disable */
/**
 * Stacks Blockchain API
 * Welcome to the API reference overview for the [Stacks Blockchain API](https://docs.hiro.so/stacks-blockchain-api).  [Download Postman collection](https://hirosystems.github.io/stacks-blockchain-api/collection.json) 
 *
 * The version of the OpenAPI document: STACKS_API_VERSION
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * meta data
 * @export
 * @interface RosettaBlockMetadata1
 */
export interface RosettaBlockMetadata1 {
    /**
     * 
     * @type {number}
     * @memberof RosettaBlockMetadata1
     */
    burn_block_height: number;
}

export function RosettaBlockMetadata1FromJSON(json: any): RosettaBlockMetadata1 {
    return RosettaBlockMetadata1FromJSONTyped(json, false);
}

export function RosettaBlockMetadata1FromJSONTyped(json: any, ignoreDiscriminator: boolean): RosettaBlockMetadata1 {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'burn_block_height': json['burn_block_height'],
    };
}

export function RosettaBlockMetadata1ToJSON(value?: RosettaBlockMetadata1 | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'burn_block_height': value.burn_block_height,
    };
}
