// @stacks/blockchain-api-client/src/generated/models/RosettaOptions.ts
/* tslint:disable */
/* eslint-disable */
/**
 * Stacks Blockchain API
 * Welcome to the API reference overview for the [Stacks Blockchain API](https://docs.hiro.so/stacks-blockchain-api).  [Download Postman collection](https://hirosystems.github.io/stacks-blockchain-api/collection.json) 
 *
 * The version of the OpenAPI document: STACKS_API_VERSION
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * The options that will be sent directly to /construction/metadata by the caller.
 * @export
 * @interface RosettaOptions
 */
export interface RosettaOptions {
    /**
     * sender's address 
     * @type {string}
     * @memberof RosettaOptions
     */
    sender_address?: string;
    /**
     * Type of operation e.g transfer
     * @type {string}
     * @memberof RosettaOptions
     */
    type?: string;
    /**
     * This value indicates the state of the operations
     * @type {string}
     * @memberof RosettaOptions
     */
    status?: string;
    /**
     * Recipient's address
     * @type {string}
     * @memberof RosettaOptions
     */
    token_transfer_recipient_address?: string;
    /**
     * Amount to be transfered.
     * @type {string}
     * @memberof RosettaOptions
     */
    amount?: string;
    /**
     * Currency symbol e.g STX
     * @type {string}
     * @memberof RosettaOptions
     */
    symbol?: string;
    /**
     * Number of decimal places
     * @type {number}
     * @memberof RosettaOptions
     */
    decimals?: number;
    /**
     * Maximum price a user is willing to pay.
     * @type {number}
     * @memberof RosettaOptions
     */
    gas_limit?: number;
    /**
     * Cost necessary to perform a transaction on the network
     * @type {number}
     * @memberof RosettaOptions
     */
    gas_price?: number;
    /**
     *  A suggested fee multiplier to indicate that the suggested fee should be scaled. This may be used to set higher fees for urgent transactions or to pay lower fees when there is less urgency.
     * @type {number}
     * @memberof RosettaOptions
     */
    suggested_fee_multiplier?: number;
    /**
     * Maximum fee user is willing to pay
     * @type {string}
     * @memberof RosettaOptions
     */
    max_fee?: string;
    /**
     * Fee for this transaction
     * @type {string}
     * @memberof RosettaOptions
     */
    fee?: string;
    /**
     * Transaction approximative size (used to calculate total fee).
     * @type {number}
     * @memberof RosettaOptions
     */
    size?: number;
    /**
     * STX token transfer memo.
     * @type {string}
     * @memberof RosettaOptions
     */
    memo?: string;
    /**
     * Number of cycles when stacking.
     * @type {number}
     * @memberof RosettaOptions
     */
    number_of_cycles?: number;
    /**
     * Address of the contract to call.
     * @type {string}
     * @memberof RosettaOptions
     */
    contract_address?: string;
    /**
     * Name of the contract to call.
     * @type {string}
     * @memberof RosettaOptions
     */
    contract_name?: string;
    /**
     * Set the burnchain (BTC) block for stacking lock to start.
     * @type {number}
     * @memberof RosettaOptions
     */
    burn_block_height?: number;
    /**
     * The reward cycle ID for stacking transaction.
     * @type {number}
     * @memberof RosettaOptions
     */
    reward_cycle_id?: number;
    /**
     * Delegator address for when calling `delegate-stacking`.
     * @type {string}
     * @memberof RosettaOptions
     */
    delegate_to?: string;
    /**
     * The reward address for stacking transaction. It should be a valid Bitcoin address
     * @type {string}
     * @memberof RosettaOptions
     */
    pox_addr?: string;
    /**
     * The hex-encoded signer key (buff 33) for PoX.
     * @type {string}
     * @memberof RosettaOptions
     */
    signer_key?: string;
    /**
     * The hex-encoded signer private key for PoX. Specify either this or `signer_signature`, otherwise the PoX transaction requires allow-listing from the signer.
     * @type {string}
     * @memberof RosettaOptions
     */
    signer_private_key?: string;
    /**
     * The hex-encoded signer signature for PoX. Specify either this or `signer_private_key`, otherwise the PoX transaction requires allow-listing from the signer.
     * @type {string}
     * @memberof RosettaOptions
     */
    signer_signature?: string;
    /**
     * The maximum amount of STX to stack for PoX. If not specified, the `amount` will be used as the `max-amount` for the PoX transaction.
     * @type {string}
     * @memberof RosettaOptions
     */
    pox_max_amount?: string;
    /**
     * The auth ID for the PoX transaction. If not specified, a random value will be generated.
     * @type {string}
     * @memberof RosettaOptions
     */
    pox_auth_id?: string;
}

export function RosettaOptionsFromJSON(json: any): RosettaOptions {
    return RosettaOptionsFromJSONTyped(json, false);
}

export function RosettaOptionsFromJSONTyped(json: any, ignoreDiscriminator: boolean): RosettaOptions {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'sender_address': !exists(json, 'sender_address') ? undefined : json['sender_address'],
        'type': !exists(json, 'type') ? undefined : json['type'],
        'status': !exists(json, 'status') ? undefined : json['status'],
        'token_transfer_recipient_address': !exists(json, 'token_transfer_recipient_address') ? undefined : json['token_transfer_recipient_address'],
        'amount': !exists(json, 'amount') ? undefined : json['amount'],
        'symbol': !exists(json, 'symbol') ? undefined : json['symbol'],
        'decimals': !exists(json, 'decimals') ? undefined : json['decimals'],
        'gas_limit': !exists(json, 'gas_limit') ? undefined : json['gas_limit'],
        'gas_price': !exists(json, 'gas_price') ? undefined : json['gas_price'],
        'suggested_fee_multiplier': !exists(json, 'suggested_fee_multiplier') ? undefined : json['suggested_fee_multiplier'],
        'max_fee': !exists(json, 'max_fee') ? undefined : json['max_fee'],
        'fee': !exists(json, 'fee') ? undefined : json['fee'],
        'size': !exists(json, 'size') ? undefined : json['size'],
        'memo': !exists(json, 'memo') ? undefined : json['memo'],
        'number_of_cycles': !exists(json, 'number_of_cycles') ? undefined : json['number_of_cycles'],
        'contract_address': !exists(json, 'contract_address') ? undefined : json['contract_address'],
        'contract_name': !exists(json, 'contract_name') ? undefined : json['contract_name'],
        'burn_block_height': !exists(json, 'burn_block_height') ? undefined : json['burn_block_height'],
        'reward_cycle_id': !exists(json, 'reward_cycle_id') ? undefined : json['reward_cycle_id'],
        'delegate_to': !exists(json, 'delegate_to') ? undefined : json['delegate_to'],
        'pox_addr': !exists(json, 'pox_addr') ? undefined : json['pox_addr'],
        'signer_key': !exists(json, 'signer_key') ? undefined : json['signer_key'],
        'signer_private_key': !exists(json, 'signer_private_key') ? undefined : json['signer_private_key'],
        'signer_signature': !exists(json, 'signer_signature') ? undefined : json['signer_signature'],
        'pox_max_amount': !exists(json, 'pox_max_amount') ? undefined : json['pox_max_amount'],
        'pox_auth_id': !exists(json, 'pox_auth_id') ? undefined : json['pox_auth_id'],
    };
}

export function RosettaOptionsToJSON(value?: RosettaOptions | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'sender_address': value.sender_address,
        'type': value.type,
        'status': value.status,
        'token_transfer_recipient_address': value.token_transfer_recipient_address,
        'amount': value.amount,
        'symbol': value.symbol,
        'decimals': value.decimals,
        'gas_limit': value.gas_limit,
        'gas_price': value.gas_price,
        'suggested_fee_multiplier': value.suggested_fee_multiplier,
        'max_fee': value.max_fee,
        'fee': value.fee,
        'size': value.size,
        'memo': value.memo,
        'number_of_cycles': value.number_of_cycles,
        'contract_address': value.contract_address,
        'contract_name': value.contract_name,
        'burn_block_height': value.burn_block_height,
        'reward_cycle_id': value.reward_cycle_id,
        'delegate_to': value.delegate_to,
        'pox_addr': value.pox_addr,
        'signer_key': value.signer_key,
        'signer_private_key': value.signer_private_key,
        'signer_signature': value.signer_signature,
        'pox_max_amount': value.pox_max_amount,
        'pox_auth_id': value.pox_auth_id,
    };
}
