// @stacks/blockchain-api-client/lib/generated/models/PoxStacker.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Stacks Blockchain API
 * Welcome to the API reference overview for the [Stacks Blockchain API](https://docs.hiro.so/stacks-blockchain-api).  [Download Postman collection](https://hirosystems.github.io/stacks-blockchain-api/collection.json)
 *
 * The version of the OpenAPI document: STACKS_API_VERSION
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PoxStackerToJSON = exports.PoxStackerFromJSONTyped = exports.PoxStackerFromJSON = exports.PoxStackerStackerTypeEnum = void 0;
/**
* @export
* @enum {string}
*/
var PoxStackerStackerTypeEnum;
(function (PoxStackerStackerTypeEnum) {
    PoxStackerStackerTypeEnum["solo"] = "solo";
    PoxStackerStackerTypeEnum["pooled"] = "pooled";
})(PoxStackerStackerTypeEnum || (exports.PoxStackerStackerTypeEnum = PoxStackerStackerTypeEnum = {}));
function PoxStackerFromJSON(json) {
    return PoxStackerFromJSONTyped(json, false);
}
exports.PoxStackerFromJSON = PoxStackerFromJSON;
function PoxStackerFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'stacker_address': json['stacker_address'],
        'stacked_amount': json['stacked_amount'],
        'pox_address': json['pox_address'],
        'stacker_type': json['stacker_type'],
    };
}
exports.PoxStackerFromJSONTyped = PoxStackerFromJSONTyped;
function PoxStackerToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'stacker_address': value.stacker_address,
        'stacked_amount': value.stacked_amount,
        'pox_address': value.pox_address,
        'stacker_type': value.stacker_type,
    };
}
exports.PoxStackerToJSON = PoxStackerToJSON;
